lib/components/viewers/route-viewer.js

```diff
@@ -10,29 +10,47 @@ import { setMainPanelContent, setViewedRoute } from '../../actions/ui'
 import { findRoutes, findRoute } from '../../actions/api'
 import { routeComparator } from '../../util/itinerary'
 
+function operatorForRoute (operators, route) {
+  return operators.find(o =>
+    o.id.toLowerCase() === route.agency.id.split(':')[0].toLowerCase())
+}
+
+function operatorIndexForRoute (operators, route) {
+  const index = operators.findIndex(o =>
+    o.id.toLowerCase() === route.agency.id.split(':')[0].toLowerCase())
+  if (index !== -1 && typeof operators[index].order !== 'undefined') return operators[index].order
+  else return 0
+}
+
 class RouteViewer extends Component {
   static propTypes = {
     hideBackButton: PropTypes.bool,
     routes: PropTypes.object
   }
 
-  _backClicked = () => {
-    this.props.setMainPanelContent(null)
-    this.props.setViewedRoute(null)
-  }
+  _backClicked = () => this.props.setMainPanelContent(null)
 
   componentWillMount () {
     this.props.findRoutes()
   }
 
   componentDidMount () {
-    //this.props.findRoute({ routeId: 'TriMet:1' })
-    //this.props.setViewedRoute({ routeId: 'TriMet:1' })
+    // this.props.findRoute({ routeId: 'TriMet:1' })
+    // this.props.setViewedRoute({ routeId: 'TriMet:1' })
   }
 
   render () {
-    const { routes, hideBackButton, languageConfig } = this.props
-
+    const { operators, routes, hideBackButton, languageConfig } = this.props
+    const sortedRoutes = routes ? Object.values(routes).sort(routeComparator) : []
+    const agencySortedRoutes = operators.length > 0
+      ? sortedRoutes.sort((a, b) => {
+        const aOperator = operatorIndexForRoute(operators, a)
```

Comment:
If `operatorIndexForRoute` always returns an integer, this can be simplified to:

```javascript
return operatorIndexForRoute(operators, a) - operatorIndexForRoute(operators, b)
```